cpp/src/knn/knn.cu

```diff
@@ -59,7 +59,7 @@ void knn_classify(cumlHandle &handle, int *out, int64_t *knn_indices,
   std::vector<int> n_unique(y.size());
 
   for (int i = 0; i < y.size(); i++) {
-    MLCommon::Label::getUniqueLabels(y[i], n_query_rows, &(uniq_labels[i]),
+    MLCommon::Label::getUniqueLabels(y[i], n_index_rows, &(uniq_labels[i]),
```

Comment:
This was an oversight originally (these variables were once very poorly named). Now that the variables are better named, it was immediately obvious to me as I was going through the code that this will causes issues.